os/fs/smartfs/smartfs_smart.c

```diff
@@ -300,6 +303,23 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 	sf->curroffset = sizeof(struct smartfs_chain_header_s);
 	sf->currsector = sf->entry.firstsector;
 	sf->byteswritten = 0;
+	
+	if ((sf->bflags & SMARTFS_BFLAG_DIRTY) == 0) {
```

Comment:
sf->bflags and sf->buffer also defined only under CONFIG_SMARTFS_USE_SECTOR_BUFFER

Reply:
yes.